ipa-core/src/net/test.rs

```diff
@@ -173,30 +328,76 @@ impl TestConfigBuilder {
                         )
                         .unwrap(),
                     ))
-                },
+                };
+                PeerConfig {
+                    url,
+                    certificate,
+                    hpke_config,
+                }
             })
-            .collect::<Vec<_>>();
-        let network = NetworkConfig::<Helper>::new_mpc(
-            peers,
-            self.use_http1
-                .then(ClientConfig::use_http1)
-                .unwrap_or_default(),
-        );
-        let servers = if self.disable_https {
-            ports.map(|ports| server_config_insecure_http(ports, !self.disable_matchkey_encryption))
-        } else {
-            HelperIdentity::make_three()
-                .map(|id| server_config_https(id, ports[id], !self.disable_matchkey_encryption))
-        };
+            .collect();
+
+        let client_config = self
+            .use_http1
+            .then(ClientConfig::use_http1)
+            .unwrap_or_default();
+        (peers, client_config)
+    }
+
+    /// Creates a test network with shards
+    #[must_use]
+    pub fn build(self) -> TestConfig {
+        // first we build all the rings and then we connect all the shards.
+        // The following will accumulate the shards as we create the rings.
+        let mut sharding_networks_shards = vec![
+            TestServers::default(),
+            TestServers::default(),
+            TestServers::default(),
+        ];
+        let rings: Vec<TestNetwork<Helper>> = (0..self.sharding_factor)
+            .map(|s| {
+                let ring_ports = self.ports_by_ring.get(s).map(|p| p.ring.clone());
+                let shards_ports = self.ports_by_ring.get(s).map(|p| p.shards.clone());
```

Comment:
```suggestion
        let r = zip(ring_ports, shard_ports).map(|(r_port, s_port)| { ... })
```

Reply:
there are cases where ports_by_ring won't be set though.